Dockerfile

```diff
@@ -173,25 +186,29 @@ RUN wget https://packages.erlang-solutions.com/erlang-solutions_1.0_all.deb \
 # Install Rust 1.51.0
 ENV RUSTUP_HOME=/opt/rust \
   PATH="${PATH}:/opt/rust/bin"
+USER dependabot
+RUN mkdir -p "$RUSTUP_HOME"
 RUN export CARGO_HOME=/opt/rust ; curl https://sh.rustup.rs -sSf | sh -s -- -y
 RUN export CARGO_HOME=/opt/rust ; rustup toolchain install 1.51.0 && rustup default 1.51.0
-
+USER root
 
 ### NEW NATIVE HELPERS
 
-COPY composer/helpers /opt/composer/helpers
-COPY dep/helpers /opt/dep/helpers
-COPY bundler/helpers /opt/bundler/helpers
-COPY go_modules/helpers /opt/go_modules/helpers
-COPY hex/helpers /opt/hex/helpers
-COPY npm_and_yarn/helpers /opt/npm_and_yarn/helpers
-COPY python/helpers /opt/python/helpers
-COPY terraform/helpers /opt/terraform/helpers
+COPY --chown=dependabot:dependabot composer/helpers /opt/composer/helpers
+COPY --chown=dependabot:dependabot dep/helpers /opt/dep/helpers
+COPY --chown=dependabot:dependabot bundler/helpers /opt/bundler/helpers
+COPY --chown=dependabot:dependabot go_modules/helpers /opt/go_modules/helpers
+COPY --chown=dependabot:dependabot hex/helpers /opt/hex/helpers
+COPY --chown=dependabot:dependabot npm_and_yarn/helpers /opt/npm_and_yarn/helpers
+COPY --chown=dependabot:dependabot python/helpers /opt/python/helpers
+COPY --chown=dependabot:dependabot terraform/helpers /opt/terraform/helpers
 
 ENV DEPENDABOT_NATIVE_HELPERS_PATH="/opt" \
   PATH="$PATH:/opt/terraform/bin:/opt/python/bin:/opt/go_modules/bin:/opt/dep/bin" \
   MIX_HOME="/opt/hex/mix"
 
+USER dependabot
```

Comment:
With deeper reflection, I wondered about the implications of this bit.

Previously, helpers would be installed and owned by the `root` user. If a downstream consumer of the image were to add a `dependabot` user to drop privileges, that user would not have write access to this code.

Now we've removed the burden of adding a `dependabot` user from the downstream consumers, but may have unintentionally made these helpers writable!

If we take [Dockerfile.development](https://github.com/dependabot/dependabot-core/blob/ac276c959e8fa193061e47420152b7e4fa178b0a/Dockerfile.development#L21-L24) as indicative of what permissions the delegated user actually needs:
* `/usr/local/.pyenv` - we address this earlier in the Dockerfile
* `/opt/rust` - we address this earlier in the Dockerfile
* `/opt/go/gopath` - this requires consideration


These make sense: python and rust may install toolchains to these locations, and Go may fetch sources to `${GOPATH}/pkg`.

---

I'm wondering if we should return to installing helpers as the `root` user, then append a `chown /opt/go/gopath` to fixup the problematic paths.

I'm not 100% sold it's necessary, just flagging the change.

Reply:
One small observation. I tried running strace on a process to see why it was taking so long to finish which I was working on a bug. strace requires root privileges and without sudo and /etc/sudoers.d this made strace unusable AFAIK.

Reply:
> I tried running strace on a process 

How did that work before? This PR doesn't change the dev shell dropping to `dependabot` [previous](https://github.com/dependabot/dependabot-core/blob/3656c3de267921c038e6d23c3055941f071e87a8/Dockerfile.development#L25)

AFAICT this capability was introduced [2018-11-21](https://github.com/dependabot/dependabot-core/commit/c7e9a9cd8715157cd86ebc7a5963ef13330dddd1) and has been broken since [2018-27-28](https://github.com/dependabot/dependabot-core/commit/63fe436b9706a63be6ef32e0cef392510028a40a)?
If this is an issue, could we address it in a separate PR?

Reply:
I attempted to undo this change and install helpers as root but was running into weird issues with composer failing during updates, so backpedalled to get it green again and test on staging

Reply:
I think this permission change is the only thing left to figure out, I ran into loads of trouble trying to untangle this without ending up with the same `chown` we set out to remove from the updater image. As some of these helpers need to have write access to the helpers anyways, e.g. `bundler`, I'm wondering if it's worth trying to restrict the rest?

Reply:
I think I got this back to where previously so the native helper scripts should not be writeable by dependabot but the `gopath` and `.bundle` folders will be without the big `chown`. This setup feels a bit gnarly but tests should fail hard if this breaks so not too worried about it accidentally breaking.